### ISSUE TITLE
Reopen log-file FIFO command does not switch over logging to newly opened logfile

### ISSUE THREAD

**Issue by karolyi** [unbit/uwsgi#2192]
Hi,

I use FreeBSD and newsyslog to rotate logfiles on my server, and I wrote a small script that will echo a `l` to the master fifo of the running UWSGI process, as defined in https://uwsgi-docs.readthedocs.io/en/latest/MasterFIFO.html?highlight=#available-commands.

Problem is, after I get the message in the newly created logfile, the logging doesn't switch over to it. Basically, the only log line I get there is:

```
Jun 25 12:13:33 jail-site-hondaforum-20200625-115556 newsyslog[66259]: logfile turned over due to -F request
[1593080013] /var/log/uwsgi/forum-prod/uwsgi.log reopened.
```
Mind you the first line there comes from newsyslog forcibly making uwsgi to switch log (testing), and the second one from UWSGI itself, and I get nothing afterwards in that file. All the logging stays in the switched logfile, in this case `uwsgi.log.0`.

Here's the instance configuration (after issuing a `s` command to the master FIFO):
```
;uWSGI instance configuration
[uwsgi]
ini = prod.ini
socket = /tmp/hondaforum-prod.sock
chmod-socket = 666
logto2 = /var/log/uwsgi/forum-prod/uwsgi.log
master-fifo = /tmp/forum-prod.master-fifo.sock
idle = 300
die-on-idle = true
chdir = /home/hondaforum-prod/project/forum-django/backend
module = forum.wsgi:application
workers = 10
threads = 10
env = DJANGO_SETTINGS_MODULE=forum.settings
env = HOME=/home/hondaforum-prod
venv = /home/hondaforum-prod/project/forum-django/venv
http-auto-gzip = true
log-slow = 5000
vacuum = true
touch-reload = /home/hondaforum-prod/project/forum-django/backend/forum/wsgi.py
memory-report = 1
thunder-lock = true
cheaper-algo = spare
cheaper = 2
cheaper-initial = 2
cheaper-step = 1
stats = /tmp/prod-stats.sock
;end of configuration
```

Basically, the way I use UWSGI is, I start up instances as vassals in emperor mode, which will then have their own master FIFOs.

This happens with the latest `master`, commit id `0a6eea89940e8270feac2d12e0e9d6c719cb1a5c`.

Please help.

**Comment by karolyi** [comment]
Adding to this ticket the information that not even a touch-reload helps. I've set up my project to do a reload with touching the wsgi.py after log rotate. According to the logs the reload succeeds, but the log file stays the switched one, that is the old with `.0` suffix added to it.

Not good.
